libs/miranum-core/src/lib/miranum-core.ts

```diff
@@ -53,39 +53,33 @@ export class MiranumCore {
         return generatedFiles;
     }
 
-    public async generateArtifact(artifactName: string, type: string, project: string): Promise<Artifact> {
-        return this.initArtifact(artifactName, type, project, this.getPathFromConfig(type));
+    public async generateArtifact(artifactName: string, type: string, projectName: string, projectPath: string): Promise<Artifact> {
+        /** checks if we have a projectConfig, whether we are on top-level of the project, and if so, if we have a type which has a subfolder
+         *  if so it initialises an artifact with the "pathInProject" set as the projectConfig.workspace defines it,
+         *  otherwise it initialises it as a standalone artifact
+         */
+        const lastFolder = projectPath.substring(projectPath.lastIndexOf("/")+1);
+        if(this.projectConfig && lastFolder == projectName && (type == "form" || type == "config" || type == "element-template")) {
+            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+            // @ts-ignore
+            return this.initArtifact(artifactName, type, projectName, this.projectConfig.workspace[`${type}s`]);
+        }
+        return this.initArtifact(artifactName, type, projectName, "");
     }
 
-    private async initArtifact(artifactName: string, type: string, project: string, basePath?: string): Promise<Artifact> {
+    /**
+     * @param artifactName: name of the artifact
+     * @param type: type of the artifact
+     * @param project: name of the project
+     * @param pathInProject: if undefined, default/base path will be used => should only be undefined to generate a project
+     * @private
+     */
+    private async initArtifact(artifactName: string, type: string, project: string, pathInProject?: string): Promise<Artifact> {
         const generator = this.generatorPlugins.get(type);
         if (!generator) {
             throw new Error(`File type ${type} is not supported.`);
         }
-        return generator.generate(artifactName, project, basePath);
-    }
-
-    private getPathFromConfig(type: string): string | undefined {
-        if (this.projectConfig) {
-            switch (type){
-                case "form": {
-                    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-                    // @ts-ignore
-                    return this.projectConfig.workspace["forms"];
-                }
-                case "config": {
-                    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-                    // @ts-ignore
-                    return this.projectConfig.workspace["processConfigs"];
-                }
-                case "element-template": {
-                    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-                    // @ts-ignore
-                    return this.projectConfig.workspace["elementTemplates"];
```

Comment:
Funktioniert das immer noch mit der aktuellen Lösung? Ich finde den aktuellen Ansatz mit `this.projectConfig.workspace[`${type}s`]` gut, aber bei element-templates geht das doch nicht, oder?

Reply:
Also bei mir funktioniert es, allerdings gebe ich dir recht - eigentlich sollte es nicht gehen ...
Schätze er nimmt unseren Ordnernamen, der is ja auch drinnen
gleiches is ja dann auch bei configs der fall - die hmüssten ja über processConfigs verfügbar sein